stl/inc/xutility

```diff
@@ -644,13 +644,6 @@ struct iterator_traits<_Ty*> {
 
 // CUSTOMIZATION POINT OBJECT ranges::iter_move
 namespace ranges {
-    // STRUCT TEMPLATE _Choice_t

```

Comment:
To be safe here, should we explicitly include `<compare>`? It is not immediately clear how those definitions end up here

Reply:
We occasionally explicitly include a header even if we get it through another path, and sometimes we comment where a piece of machinery lives. Looking at this again, I think that it's just weird for these helper templates to live in `<compare>` even though it happens to be the first place where they're used. As the throughput implications are near-zero and we'll probably need them elsewhere in the future, I will promote them all the way up to `<concepts>` (as they are concept helpers) and that way I don't think we need either explicit includes or comments.